repos/system_upgrade/el7toel8/actors/checkrhui/actor.py

```diff
@@ -0,0 +1,59 @@
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.models import (
+    InstalledRPM,
+    RHUIInfo,
+    RequiredTargetUserspacePackages,
+    DNFPluginInfo
+)
+from leapp.reporting import Report, create_report
+from leapp import reporting
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common import rhsm, rhui
+
+
+class CheckRHUI(Actor):
+    """
+    Check if system is using RHUI infrastructure (on public cloud) and send messages to
+    provide additional data needed for upgrade.
+    """
+
+    name = 'checkrhui'
+    consumes = (InstalledRPM)
+    produces = (RHUIInfo, RequiredTargetUserspacePackages, Report, DNFPluginInfo)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
```

Comment:
due to testability please put the content in a library function so it's testable - Thanks.